### ISSUE TITLE
[QUESTION]How can I confirm that I have completed one epoch in Megtron

### ISSUE THREAD

**Issue by Xiaotianhe123** [NVIDIA/Megatron-LM#3012]
How can I confirm that I have completed one epoch in Megtron ,   
This is a pretrain task , i have the file idx,  bin ,  and my train config as : 
TRAINING_ARGS=(
      --training-phase pretrain
      --seq-length 16384
      --max-position-embeddings 16384
      --init-method-std 0.006
      --micro-batch-size 1
      --global-batch-size 2048
      --lr 2.0e-6
      --min-lr 1.0e-6
      --clip-grad 1.0
      --weight-decay 0.1
      --optimizer adam
      --adam-beta1 0.9
      --adam-beta2 0.95
      --adam-eps 1e-08
      --norm-epsilon 1e-6
      --train-iters 5000
      --lr-decay-iters 5000
      --lr-decay-style cosine
      --lr-warmup-fraction 0.002
      --initial-loss-scale 65536
      --bf16
      --load ${CHECKPOINT_PATH}
      --save ${CHECKPOINT_PATH}
      --save-interval 200
      --eval-interval 100
      --eval-iters 10
      --recompute-granularity full
      --recompute-method block
      --recompute-num-layers 24
    )

how can i confirm consumed samples, to finish one epoch datasets

**Comment by maanug-nv** [comment]
Hi @Xiaotianhe123 . When the training run starts, before the first iteration the dataloader setup should log how many total samples are in your dataset. You can then set `--train-samples` to that value or calculate and set `--train-iters` based on GBS.
Hope that helps.

**Comment by maanug-nv** [comment]
@Xiaotianhe123 please let us know if this is resolved for you know.
